xbmc/addons/settings/AddonSettings.cpp

```diff
@@ -249,7 +249,9 @@ bool CAddonSettings::Load(const CXBMCTinyXML& doc)
       auto settingId = categoryId;
       if (!settingId.empty())
         settingId += ".";
-      settingId += setting->ValueStr();
+      auto id = setting->ToElement()->Attribute("id");
+      if (id)
+        settingId += id;
```

Comment:
Maybe this should be 

```c++
auto id = setting->ToElement()->Attribute("id");
if (id)
{
  if (!settingId.empty())
    settingId += ".";
  settingId += id;
}
```

to avoid the dangling "."?